src/jit/hwintrinsicxarch.cpp

```diff
@@ -234,6 +234,54 @@ int Compiler::numArgsOfHWIntrinsic(NamedIntrinsic intrinsic)
     return hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].numArgs;
 }
 
+//------------------------------------------------------------------------
+// numArgsOfHWIntrinsic: get the number of arguments based on HW GenTree node
```

Comment:
The description here is incorrect. It is only based on the node if the table entry is -1

Reply:
Not really, it retrieves `numArgs` when table entry is either >= 0 of -1. So essentially it gets the exact number of args for the given intrinsic when it is needed.

Reply:
I will try to improve description to emphasize this behavior.

Reply:
> Not really, it retrieves numArgs when table entry is either >= 0 or -1.

The code is doing exactly this:
```
int numArgs = hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].numArgs;
if (numArgs != -1)
{
    return numArgs;
}
```

So it is first checking the table and returns that value if the entry wasn't `-1`. Otherwise, it checks the node operands and returns a value based on those.

Reply:
So, if someone had a table entry of `-2`, this function would return back `-2`.